File: app/channels/questions_channel.rb
class QuestionsChannel < ApplicationCable::Channel
  def subscribed
    # stream_from "some_channel"
    stream_from 'questions'
  end

  def unsubscribed
    # Any cleanup needed when channel is unsubscribed
  end

  # def follow
  #   stream_from 'questions'
  # end
end

File: app/javascript/channels/questions_channel.js
import consumer from "./consumer";

consumer.subscriptions.create({ channel: 'QuestionsChannel' }, {
  connected() {
    // Called when the subscription is ready for use on the server
    console.log('connected!');
    // console.log(gon.question_id);
  },

  disconnected() {
    // Called when the subscription has been terminated by the server
    console.log('disconnected!');
  },

  received(data) {
    console.log('received', data);
    // Called when there's incoming data on the websocket for this channel
    this.addQuestion(data);
  },

  addQuestion(html) {
    const questionList = document.querySelector('.questions');
    questionList.insertAdjacentHTML('beforeEnd', html);
  }
});
